src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java

```diff
@@ -294,23 +289,66 @@ public void createManualProgrammingExerciseResult() throws Exception {
         assertThat(response.getFeedbacks().size()).isEqualTo(result.getFeedbacks().size());
     }
 
+    @Test
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_manualResultsNotAllowed() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.AUTOMATIC);
+        Result result = ModelFactory.generateResult(true, 200).resultString("Good effort!");
+        List<Feedback> feedbacks = ModelFactory.generateFeedback().stream().peek(feedback -> feedback.setText("Good work here")).collect(Collectors.toList());
+        result.setFeedbacks(feedbacks);
+        result.setParticipation(participation);
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
+        when(gitService.getLastCommitHash(ArgumentMatchers.any())).thenReturn(ObjectId.fromString(dummyHash));
```

Comment:
wrong style, please use `doReturn...when`

Make sure to check other tests before implementing something wrong

Reply:
Well I used the current test implementation. But yes I switched to this syntax. I see it offers a bit better type check.